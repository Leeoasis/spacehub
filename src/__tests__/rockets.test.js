import React from 'react';
import renderer from 'react-test-renderer';
import Rocket from '../components/rockets';

describe('Rocket component', () => {
  const rocket = {
    id: 'falcon1',
    name: 'Falcon 1',
    description: 'Small launch vehicle',
    flickr_images: ['https://www.example.com/falcon1.jpg'],
    reserved: true,
  };

  it('renders correctly when rocket is reserved', () => {
    const tree = renderer
      .create(<Rocket rocket={rocket} handleReserveClick={() => {}} onCancelClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when rocket is not reserved', () => {
    const newRocket = { ...rocket, reserved: false };
    const tree = renderer
      .create(<Rocket rocket={newRocket} handleReserveClick={() => {}} onCancelClick={() => {}} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
