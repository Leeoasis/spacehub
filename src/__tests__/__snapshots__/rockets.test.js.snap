// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component renders correctly when rocket is not reserved 1`] = `
<div
  className="Rocket-Card"
>
  <div
    className="Rocket-Images"
  >
    <img
      alt="Rocket Falcon 1"
      src="https://www.example.com/falcon1.jpg"
    />
  </div>
  <div>
    <h2>
      Falcon 1
    </h2>
    <p
      className="card-text"
    >
      <span
        className="hidden"
      >
        Reserved
      </span>
      Small launch vehicle
    </p>
    <button
      className="Reserve-Button"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</div>
`;

exports[`Rocket component renders correctly when rocket is reserved 1`] = `
<div
  className="Rocket-Card"
>
  <div
    className="Rocket-Images"
  >
    <img
      alt="Rocket Falcon 1"
      src="https://www.example.com/falcon1.jpg"
    />
  </div>
  <div>
    <h2>
      Falcon 1
    </h2>
    <p
      className="card-text"
    >
      <span
        className="show"
      >
        Reserved
      </span>
      Small launch vehicle
    </p>
    <button
      className="Cancel-Reservation"
      onClick={[Function]}
      type="button"
    >
      Cancel Reservation
    </button>
  </div>
</div>
`;
